src/main/java/uk/gov/ons/ctp/common/rest/RestClient.java

```diff
@@ -144,19 +144,49 @@ public RestTemplate getRestTemplate() {
       MultiValueMap<String, String> queryParams,
       Object... pathParams)
       throws ResponseStatusException {
-    log.debug("Enter getResources for path: {}", path);
+    return doHttpOperation(
+        HttpMethod.GET, path, null, clazz, headerParams, queryParams, pathParams);
+  }
+
+  /**
+   * Use to perform an http operation GET/PUT/POST to retrieve a single resource
+   *
+   * @param <T> the type that will returned by the server we call
+   * @param <O> is the type of payload to send.
+   * @param method is the type of http call to be made.
+   * @param path the API path - can contain path params place holders in "{}" ie "/cases/{caseid}"
+   * @param objToSend is the object to send as a Put or Post payload.
+   * @param clazz the class type of the resource to be obtained
+   * @param headerParams map of header of params to be used - can be null
+   * @param queryParams multi map of query params keyed by string logically allows for
+   *     K:"haircolor",V:"blond" AND K:"shoesize", V:"9","10"
+   * @param pathParams vargs list of params to substitute in the path - note simply used in order
+   * @return the type you asked for! or null
+   * @throws ResponseStatusException something went wrong making http call. The reason field will contain the http response body.
+   */
+  @SuppressWarnings("unchecked")
```

Comment:
No need to suppress once the object mapper stuff is removed.

Reply:
Removed